### PR TITLE
Bug 1450018 - Convert Similar jobs tab to ReactJS

### PR DESCRIPTION
**Pull request by camd**
Replaces the Similar Jobs panel with a ReactJS version.  I took out the filtering by Machine.  I also used a slightly shorter date format for the list of similar jobs so that it could (at least sometimes) show on one line.  Hovering shows the exact date.  But otherwise it should be the same as before.

### CODE REVIEW

File: ui/details-panel/SimilarJobsTab.jsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
+import PropTypes from 'prop-types';
```
Comment (edmorley) [inline ui/details-panel/SimilarJobsTab.jsx]: Could you add a `import React from 'react';` at the top of this file, so it doesn't cause linter errors when #3481 is merged?

File: ui/details-panel/SimilarJobsTab.jsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+      // get pushes and revisions for the given ids
+      const pushListResp = await this.ThResultSetModel.getResultSetList(repoName, pushIds, true);
+      const pushList = pushListResp.data;
+      //decorate the list of jobs with their result sets
+      const pushes = pushList.results.reduce((acc, push) => (
+        { ...acc, [push.id]: push }
+      ), {});
+      newSimilarJobs.forEach((simJob) => {
+        simJob.result_set = pushes[simJob.result_set_id];
+        simJob.revisionResultsetFilterUrl = `/#/jobs?repo=${repoName}&revision=${simJob.result_set.revisions[0].revision}`;
```
Comment (edmorley) [inline ui/details-panel/SimilarJobsTab.jsx]: Perhaps these could be generated via a modified `jobsUrl()` in `urlHelper.js`?

File: ui/entry-index.js [hunk 3]
```diff
@@ -68,9 +68,9 @@ import './js/controllers/tcjobactions';
 import './plugins/tabs';
 import './plugins/controller';
 import './plugins/pinboard';
-import './plugins/similar_jobs/controller';
```
Comment (edmorley) [inline ui/entry-index.js]: The `ui/plugins/similar_jobs/controller.js` file can now be removed too :-)

File: ui/details-panel/SimilarJobsTab.jsx (hunk 1 above)
Comment (camd) [reply ui/details-panel/SimilarJobsTab.jsx]: good call, done.

File: ui/details-panel/SimilarJobsTab.jsx (hunk 2 above)
Comment (camd) [reply ui/details-panel/SimilarJobsTab.jsx]: Another good call.   Done

File: ui/entry-index.js (hunk 3 above)
Comment (camd) [reply ui/entry-index.js]: Wow, talk about a good call:  This is one.  done.  :)

**Pull request merged**